include/aws/s3/private/s3_util.h

```diff
@@ -165,6 +165,29 @@ void aws_s3_get_part_range(
     uint64_t *out_part_range_start,
     uint64_t *out_part_range_end);
 
+/**
+ * TODO: Move the function below to aws-c-common. This is a copy from aws-c-http/include/private/strutil.h.
+ *
+ * Read entire cursor as ASCII/UTF-8 unsigned base-10 number.
+ * Stricter than strtoull(), which allows whitespace and inputs that start with "0x"
+ *
+ * Examples:
+ * "0" -> 0
+ * "123" -> 123
+ * "00004" -> 4 // leading zeros ok
+ *
+ * Rejects things like:
+ * "-1" // negative numbers not allowed
+ * "1,000" // only characters 0-9 allowed
+ * "" // blank string not allowed
+ * " 0 " // whitespace not allowed
+ * "0x0" // hex not allowed
+ * "FF" // hex not allowed
+ * "999999999999999999999999999999999999999999" // larger than max u64
+ */
+AWS_S3_API
```

Comment:
Pretty sure this will break on certain compilers depending on linker flags due to symbol conflicts. I’d in-line it where you need it until it’s moved to aws-c-common

Reply:
Renamed to `aws_s3_strutil_read_unsigned_num` to avoid name clash with `aws_s3_strutil_read_unsigned_num`, until the refactor is done. As part of the refactor task I'll delete the function definition in `aws-c-http` and this corresponding copy `aws_s3_strutil_read_unsigned_num` in `aws-c-s3`, and fix callers of the function in `aws-c-http` and `aws-c-s3` so they use the version from `aws-c-common`.